app/src/main/java/com/somestupidappproject/betago/utils/BoardUtils.java

```diff
@@ -36,12 +37,38 @@
         return ret;
     }
 
+    public static ArrayList<Stone> getDeadStones(Stone stone, Board board) {
+        ArrayList<Stone> deadStones = new ArrayList<>();
+        if (stone.getColor() == Stone.UNTAKEN) {
+            return deadStones;
+        }
+
+        // Get all neighbors and also check if the placed point is surrounded
+        Set<Stone> neighbors = BoardUtils.getNeighbors(stone, board);
+
+        for (Stone neighbor : neighbors) {
+            if (!LogicUtil.isAlive(neighbor, board)) {
+                // Point is dead which means it's group is also dead
+                // Find the group here
+                Set<Stone> group = LogicUtil.findGroup(neighbor, board);
+                deadStones.addAll(group);
+            }
+        }
+
+        return deadStones;
+    }
+
     // Determines if the move is valid or not
     // invalid moves included are currently
     // 1. Suicide move aka going in a spot where the group you are adding to has no liberties
-    // 2. NOT IMPLEMENTED Ko invalid move
+    // 2. Ko invalid move
     // 3. Occupying a spot that already has a piece
-    public static boolean isValidMove(Stone stone, Board board, boolean isBlacksMove) {
+    public static boolean isValidMove(
+            Stone stone,
+            Board board,
+            boolean isBlacksMove,
+            ArrayList<Stone> previousCapturedStones,
+            Stone previousStone ) {
         // convert the boolean into an int
         int color = isBlacksMove ? 1 : 2;
```

Comment:
this looks like an unused variable, plus stone stores the color information already